components/automate-ui/src/app/modules/desktop/desktop-detail/desktop-detail.component.ts

```diff
@@ -103,17 +76,18 @@ export class DesktopDetailComponent implements OnInit, OnDestroy {
     this.downloadDropdownVisible = false;
   }
 
-  addCheckInLabels(checkInHistory: DailyNodeRunsStatus[]): DailyNodeRunsStatus[] {
-    let numWeeks = Math.floor(checkInHistory.length / 7);
-    checkInHistory.forEach((history: DailyNodeRunsStatus, index: number) => {
+  get labeledCheckInHistory() {
+    const buckets = this.checkInHistory.durations.buckets;
```

Comment:
I was wondering how we know to divide buckets by 7...would suggest add a comment somewhere in here explaining that a bucket contains one day of stuff.

Reply:
Tara originally wrote this method so I may misspeak; but I believe the intention was to determine the number of weeks within the returned array of days. I'd like to push this logic into a reducer somewhere but I think this okay for now.